pluginupdate.py: add support for adding/updating individual plugins (#336137)
NixOS/nixpkgs: doc/languages-frameworks/vim.section.md

Alternatively, set the number of processes to a lower count to avoid rate-limiting.

```sh
nix-shell -p vimPluginsUpdater --run 'vim-plugins-updater --proc 1'
```

> If you want to update only certain plugins, you can specify their names to `update` flag. Note that those name must be the same as in `pkgs/applications/editors/vim/plugins/vim-plugin-names` file.

Reviewer: ```suggestion
If you want to update only certain plugins, you can specify them after the `update` command. Note that you must use the same plugin names as the `pkgs/applications/editors/vim/plugins/vim-plugin-names` file.
```

> If you want to update only certain plugins, you can specify them after the `update` command. Note that you must use the same plugin names as the `pkgs/applications/editors/vim/plugins/vim-plugin-names` file.

Reviewer 2: we can remove this since single plugin update will be in a separate PR ?

Author: Done in #353786!

```sh

> nix-shell -p vimPluginsUpdater --run 'vim-plugins-updater --update "nvim-treesitter" "LazyVim"'

Reviewer 2: apparently it's not --update but update:
```
nix run .#vimPluginsUpdater --  update nvim-treesitter LazyVim
WARNING:root:You have enabled parallel updates but haven't set a github token.
You may be hit with `HTTP Error 429: too many requests` as a consequence.Either set --proc=1 or --github-token=YOUR_TOKEN. 
path is '/nix/store/rgwip825v0m4nhsvqyabf0mb2as81x0q-0b8b78f9d08dc338a146eb4cd4bcbed8dd36a783.tar.gz'
2 of 1471 were checked
```

Author: Fixed [here](https://github.com/NixOS/nixpkgs/compare/0c322416d8569700c1a24589e060a04243eb581c..b306fc0cf6fc572163b1da98834def72621c9432), thanks!

Alternatively, set the number of processes to a lower count to avoid rate-limiting.

```sh
nix-shell -p vimPluginsUpdater --run 'vim-plugins-updater --proc 1'
```

If you want to update only certain plugins, you can specify them after the `update` command. Note that you must use the same plugin names as the `pkgs/applications/editors/vim/plugins/vim-plugin-names` file.

```sh

> nix-shell -p vimPluginsUpdater --run 'vim-plugins-updater update "nvim-treesitter" "LazyVim"'

Author: No, but it is easier to read if they are there. I can remove it if you want

> nix-shell -p vimPluginsUpdater --run 'vim-plugins-updater update "nvim-treesitter" "LazyVim"'

Reviewer 3: I don't have a strong feeling, we can keep them.
